runtime/src/main/java/org/corfudb/protocols/CorfuProtocolCommon.java

```diff
@@ -0,0 +1,213 @@
+package org.corfudb.protocols;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ByteString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.SequencerMetrics;
+import org.corfudb.protocols.wireprotocol.StreamAddressRange;
+import org.corfudb.protocols.wireprotocol.StreamsAddressResponse;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.exceptions.SerializerException;
+import org.corfudb.runtime.proto.RpcCommon.LayoutMsg;
+import org.corfudb.runtime.proto.RpcCommon.SequencerMetricsMsg;
+import org.corfudb.runtime.proto.RpcCommon.SequencerMetricsMsg.SequencerStatus;
+import org.corfudb.runtime.proto.RpcCommon.StreamAddressRangeMsg;
+import org.corfudb.runtime.proto.RpcCommon.StreamAddressSpaceMsg;
+import org.corfudb.runtime.proto.RpcCommon.TokenMsg;
+import org.corfudb.runtime.proto.RpcCommon.UuidMsg;
+import org.corfudb.runtime.proto.RpcCommon.UuidToStreamAddressSpacePairMsg;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.stream.StreamAddressSpace;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * This class provides methods for creating and converting between the Protobuf
+ * objects defined in rpc_common.proto and their Java counterparts. These are used
+ * by the majority of the service RPCs.
+ */
+@Slf4j
+public class CorfuProtocolCommon {
+    // Prevent class from being instantiated
+    private CorfuProtocolCommon() {}
+
+    private static final EnumMap<SequencerMetrics.SequencerStatus, SequencerStatus> sequencerStatusTypeMap =
+            new EnumMap<>(ImmutableMap.of(
+                    SequencerMetrics.SequencerStatus.READY, SequencerStatus.READY,
+                    SequencerMetrics.SequencerStatus.NOT_READY, SequencerStatus.NOT_READY,
+                    SequencerMetrics.SequencerStatus.UNKNOWN, SequencerStatus.UNKNOWN));
+
+    // Temporary message header markers indicating message type.
+    public static final byte LEGACY_CORFU_MSG_MARK = 0x1;
```

Comment:
Would be great to use enum instead of static variables, like:
```
enum MessageMarker {
  LEGACY_CORFU_MSG_MARK(0x1), PROTO_CORFU_REQUEST_MSG_MARK(0x2), PROTO_CORFU_RESPONSE_MSG_MARK(0x3)

  private final byte value;
}
```

Reply:
Also you don't need "CORFU" prefix in their names (it reduces length a bit)

Reply:
Done.